lib/web_ui/lib/src/engine/view_embedder/hot_restart_cache_handler.dart

```diff
@@ -11,10 +13,14 @@ import '../dom.dart';
 /// to clear.  Delay removal of old visible state to make the
 /// transition appear smooth.
 @JS('window.__flutterState')
-external List<Object?>? get hotRestartStore;
+external JSArray? get _hotRestartStore;
+List<Object?>? get hotRestartStore =>
+    _hotRestartStore?.toObjectShallow as List<Object>?;
 
 @JS('window.__flutterState')
-external set hotRestartStore(List<Object?>? nodes);
+external set _hotRestartStore(JSArray? nodes);
```

Comment:
I think this one is a prime candidate for eliminating all the `toDart` conversion stuff and using JS types directly. It's localized to just this file, so it should be a pretty straightforward change. Any chance you might actually switch this over Joshua? I think it would be really nice as a small example we can reference for places where we directly use the JS objects themselve.

Reply:
Ah, great call! IIRC this JSArray is never used from JS actually, so the shallower you can make this conversion, the better :P

Reply:
Actually, I'm just going to revert this change. Until we reify JS types on JS backends, it's just too awkward to use the JS types past the shim layer.

The one notable exception to this policy is `JSFunction`, because we really shouldn't be calling JS functions from Dart anyways.

Anyway, RE: JSArray, let's circle back after reification and I can write some canonical example code, for now we'll just convert these behind the scenes.